platform-tools-sdk/cargo-build-sbf/src/toolchain.rs

```diff
@@ -569,3 +578,147 @@ pub(crate) fn rust_target_triple(config: &Config) -> String {
         format!("sbpf{}-solana-solana", config.arch)
     }
 }
+
+fn fix_all_bins_and_dylibs(path: &Path) -> Result<(), std::io::Error> {
+    for libdir in [path.join("llvm/lib"), path.join("rust/lib")] {
+        for candidate in std::fs::read_dir(libdir)? {
+            let Ok(candidate) = candidate else { continue };
+            if path_is_dylib(&candidate.path()) {
+                fix_bin_or_dylib(path, &candidate.path());
+            }
+        }
+    }
+    for bindir in [path.join("llvm/bin"), path.join("rust/bin")] {
+        for candidate in std::fs::read_dir(bindir)? {
+            let Ok(candidate) = candidate else { continue };
+            fix_bin_or_dylib(path, &candidate.path());
+        }
+    }
+    for targetdir in std::fs::read_dir(path.join("rust/lib/rustlib"))? {
+        let targetdir = targetdir?;
+        for bindir in ["bin", "bin/gcc-ld"] {
+            let Ok(bindir_candidates) = std::fs::read_dir(targetdir.path().join(bindir)) else {
+                continue;
+            };
+            for candidate in bindir_candidates {
+                let Ok(candidate) = candidate else { continue };
+                fix_bin_or_dylib(path, &candidate.path());
+            }
+        }
+    }
+    Ok(())
+}
+
+fn fix_bin_or_dylib(out: &Path, fname: &Path) {
+    debug!("attempting to patch {}", fname.display());
+    // Only build `.nix-deps` once.
+    static NIX_DEPS_DIR: OnceLock<PathBuf> = OnceLock::new();
+    let mut nix_build_succeeded = true;
+    let nix_deps_dir = NIX_DEPS_DIR.get_or_init(|| {
+        // Run `nix-build` to "build" each dependency (which will likely reuse the existing
+        // `/nix/store` copy, or at most download a pre-built copy).
+        //
+        // Importantly, we create a gc-root called `.nix-deps` in the target directory, but still
+        // reference the actual `/nix/store` path in the rpath as it makes it significantly more
+        // robust against changes to the location of the `.nix-deps` location.
+        //
+        // bintools: Needed for the path of `ld-linux.so` (via `nix-support/dynamic-linker`).
+        // zlib: Needed as a system dependency of various LLVM tools.
+        // patchelf: Needed for patching ELF binaries.
+        // libgcc.lib: libstdc++
+        let nix_deps_dir = out.join(".nix-deps");
+        const NIX_EXPR: &str = "
+        with (import <nixpkgs> {});
+        symlinkJoin {
+            name = \"solana-sbf-dependencies\";
+            paths = [
+                zlib
+                patchelf
+                stdenv.cc.bintools
+                libgcc.lib
```

Comment:
This is my own lack of understanding, but the nix setup I've been using patches with `libclang.lib` instead: https://github.com/arijoon/solana-nix/blob/1f162a3cc6af49b6437debcee850a632297b8f8c/solana-platform-tools.nix#L76

Are they interchangeable in this case and it doesn't make a big deal? If so I'd lean towards keeping it all using clang stuff, rather than mixing with gcc. But again, I might be missing something